newrelic/api/ml_model.py

```diff
@@ -52,3 +53,35 @@ def get_ai_message_ids(response_id=None):
         return [{"conversation_id": conversation_id, "request_id": request_id, "message_id": _id} for _id in ids]
     warnings.warn("No message ids found. get_ai_message_ids must be called within the scope of a transaction.")
     return []
+
+
+def record_ai_feedback_event(
+    message_id, rating, conversation_id=None, request_id=None, category=None, message=None, metadata=None
+):
+    transaction = current_transaction()
+    if not transaction:
+        warnings.warn(
+            "No message feedback events will be recorded. record_ai_feedback must be called within the "
+            "scope of a transaction."
+        )
+        return
+
+    feedback_message_id = str(uuid.uuid4())
+    try:
+        rating = str(rating)
+    except Exception:
+        pass
+
+    feedback_message_event = {
+        "id": feedback_message_id,
+        "message_id": message_id,
+        "rating": rating,
+        "metadata": metadata or "",
```

Comment:
I'd expect metadata to be dictionary of extra user attributes to attach to the event so it seems like it would be better to attach the metadata so that it appears at the same level as the rest of the attributes:
```
metadata = metadata or {}
feedback_message_event.update(metadata)
```